code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -972,57 +973,82 @@ func (sc *StorageSmartContract) populateGenerateChallenge(
 		return nil, errors.New("invalid blobber for allocation")
 	}
 
+	if alloc.DataShards+1 > minValidators {
+		minValidators = alloc.DataShards + 1
+	}
+
 	var randValidators []ValidationPartitionNode
 	if err := validators.GetRandomItems(balances, r, &randValidators); err != nil {
 		return nil, common.NewError("add_challenge",
 			"error getting validators random slice: "+err.Error())
 	}
 
+	if len(randValidators) < minValidators {
+		return nil, errors.New("validators number does not meet minimum challenge requirement")
+	}
+
 	var (
-		needValidNum       = minInt(len(randValidators), alloc.DataShards+1)
-		selectedValidators = make([]*ValidationNode, 0, needValidNum)
-		perm               = r.Perm(len(randValidators))
+		selectedValidators  = make([]*ValidationNode, 0, minValidators)
+		perm                = r.Perm(len(randValidators))
+		remainingValidators = len(randValidators)
 	)
 
 	now := txn.CreationDate
 	filterValidator := filterHealthyValidators(now)
 
-	for i := 0; i < needValidNum; i++ {
+	for i := 0; i < len(randValidators) && len(selectedValidators) < minValidators; i++ {
+		if remainingValidators < minValidators {
```

Comment:
imagine we have 
```go
minValidators = 50
remainingValidators = 50 //partitionSize
```
we still can fill the list of validators, also we do not decrease the minValidators on every successful validator, which doesn;'t look correct, I'd suggest to remove this check

Reply:
I don't understand the logic, how can you address the situation in my comment?

Reply:
> I don't understand the logic, how can you address the situation in my comment?

_As per current implementation it would work this way:_  if the `minValidators = 50` and `remainingValidators = 50`, loop will iterate through all the validators in the list checking if they meet the criteria, and if any validator gets kicked (filtered out) it will stop challenge creation, because the remainingValidators will be decremented and become 49 which will fail `remainingValidators < minValidators` check on next iteration.  

I will change the code to only get certain number of validators as set by `validators_per_challenge` in config

Reply:
@din-mukhammed let's select fixed number (validators_per_challenge) validators.